### PR TITLE
grpc: add dashboard for site-configuration service to frontend grafana page

### PR DESCRIPTION
**Pull request by ggilmore**
This PR adds a gRPC grafana dashboard for frontend's site configuration server. This required some refactoring to support multiple gRPC service dashboards on the same grafana page.

screencapture-localhost-3370-debug-grafana-d-frontend-frontend-2023-09-19-15_04_01

## Test plan

CI

Produced the above screenshot while connected to sourcegraph.com's prometheus instance:

## Preview 🤩
[Preview Link](https://docs.sourcegraph.com/@grpc-frontend-2-servers)

### CODE REVIEW

**Review by efritz — APPROVED** [review]

File: monitoring/definitions/shared/grpc.go [hunk 1]
```diff
@@ -39,6 +42,8 @@ type GRPCServerMetricsOptions struct {
 // NewGRPCServerMetricsGroup creates a group containing statistics (request rate, request duration, etc.) for the grpc service
 // specified in the given opts.
 func NewGRPCServerMetricsGroup(opts GRPCServerMetricsOptions, owner monitoring.ObservableOwner) monitoring.Group {
+	opts.HumanServiceName = strcase.ToSnake(opts.HumanServiceName)
```
Comment (eseliger) [inline monitoring/definitions/shared/grpc.go:45]: could we make this an exported func instead? https://github.com/sourcegraph/sourcegraph/blob/b9875f311fa9fb9915e960210ba31477c51c48fd/internal/observation/snakecase.go#L9

File: schema/schema.go
Comment (eseliger) [inline schema/schema.go:1]: are the changes in this file related?

File: schema/schema.go [hunk 2]
```diff
@@ -1692,14 +1692,15 @@ type OnboardingStep struct {
 	Info string `json:"info,omitempty"`
 	// Label description: Label of the step shown to the user
 	Label string `json:"label"`
+	// RequriredSteps description: Set this property if only a subset of steps are required for this task to complete.
+	RequriredSteps float64 `json:"requriredSteps,omitempty"`
```
Comment (eseliger) [inline schema/schema.go:1696]: typo? 🤔

**Review by eseliger — APPROVED** [review]

**Comment by ggilmore** [comment]
@cla-bot check

File: monitoring/definitions/shared/grpc.go (hunk 1 above)
Comment (ggilmore) [reply monitoring/definitions/shared/grpc.go:45]: I can do that in a follow up PR.

File: schema/schema.go
Comment (ggilmore) [reply schema/schema.go:1]: It looks like we don't check in CI that `bazel run //dev:write_all_generated` doesn't produce any changes.

**Pull request merged**